gaia-header.js

```diff
@@ -145,13 +327,172 @@ module.exports = Component.register('gaia-header', {
    * @param  {Event} e
    * @private
    */
-  onActionButtonClick: function(e) {
-    var config = { detail: { type: this.getAttribute('action') } };
-    var actionEvent = new CustomEvent('action', config);
-    setTimeout(this.dispatchEvent.bind(this, actionEvent));
+  onActionButtonClick: function() {
+    debug('action button click');
+    var config = { detail: { type: this.action } };
+    var e = new CustomEvent('action', config);
+    setTimeout(() => this.dispatchEvent(e));
+  },
+
+  /**
+   * Get the margin required to center
+   * the title between surrounding buttons.
+   *
+   * We don't do any DOM writes in this function
+   * so that we can batch them together with
+   * other write for optimum performance.
+   *
+   * @param  {Object} title
+   * @return {Object}
+   * @private
+   */
+  getTitleMarginStart: function() {
+    var titleStart = this.titleStart;
+    var titleEnd = this.titleEnd;
+    debug('get title margin', titleStart, titleEnd);
+    return titleStart === titleEnd ? 0 : titleEnd - titleStart;
```

Comment:
I don't really get this ternary operator :) why don't you simply return `titleEnd - titleStart`? If they're equal values, it should still be 0 :D